provider/manifest/watchdog.go

```diff
@@ -46,7 +50,11 @@ func newWatchdog(sess session.Session, parent <-chan struct{}, done chan<- dtype
 }
 
 func (wd *watchdog) stop() {
-	wd.lc.ShutdownAsync(nil)
+	select {
+	case <-wd.stopLock:
+		wd.lc.ShutdownAsync(nil)
```

Comment:
I think this is a good band-aid, but a band-aid nonetheless.

If a second `stop()` is blocking, we should try to figure out why.

Reply:
Why is it a band aid? I thought the lifecycle component was never meant to be called twice for shutdown?

Reply:
`ShutdownAsync` can be called twice if it is not being called from the same thread as the main loop and `ShutdownInitiated()` is called quickly after the first one like is being done here: https://github.com/ovrclk/akash/blob/6184be05aa61bc2059b23dc2a72147de9180ba68/provider/manifest/watchdog.go#L59-L60

Reply:
I see - if [this](https://github.com/ovrclk/akash/blob/6184be05aa61bc2059b23dc2a72147de9180ba68/provider/manifest/watchdog.go#L64-L67) is blocking then even the first call will block.

Reply:
Yes, that is one case. I could change it so that the context used is tied into the lifecycle object and closed when the lifecycle object starts closing. I'm not entirely sure if that is desirable, it might actually mean the bid gets left open however.

Reply:
I think using `runner.Run()` to execute the tx asynchronously while still handling shutdowns is good enough.  wait for the runner to finish before returning if it has already started.

Reply:
OK, let me take a pass at that.